lib/modules/apostrophe-ui/public/js/ui.js

```diff
@@ -25,6 +25,66 @@ apos.define('apostrophe-ui', {
 
     self.options = options;
 
+    // Utility that lets you add a tooltip to any HTML element using
+    // the data-apos-tooltip="YR TOOLTIP MSG" pattern
+
+    // If using outside of the current implementation (widget controls and chooser controls)
+    // you need assign your own event handlers
+
+    self.prepTooltip = function() {
+      var $tooltip = $('[data-apos-tooltip-template]');
+      var $clone = $tooltip.clone();
+      var $body = $('body');
+      return $clone
+        .removeAttr('data-apos-tooltip-template')
+        .attr('data-apos-tooltip', 'true')
+        .removeClass('apos-hidden');
+    };
+
+    self.createTooltip = function($el) {
+      var delay = 500;
+      var $body = $('body');
+      var $this = $(this);
+      var $newTooltip = self.prepTooltip().clone();
+      var left = $this.offset().left;
+      var top = $this.offset().top;
+      var width = $this.outerWidth();
+      var height = $this.outerHeight();
+
+      window.apos.tooltipDelayTimeout = setTimeout(function () {
```

Comment:
I'm fine with this approach. The apos namespace is ours and you namespaced the name further.

Reply:
(Well mostly ours, but nobody's going to alias an apostrophe module as "tooltipDelayTimeout.")